### ISSUE TITLE
Can't control DateTime representations in `Example`

### ISSUE THREAD

**Issue by masters3d** [commandlineparser/commandline#258]
code:
```csharp
    class Options
    {
        [Option("startDate", Required = true, HelpText = "The start of the date range. yyyy-mm-dd")]
        public DateTime StartDate { get; set; }

        [Option("endDate", Required = true, HelpText = "The end of the date range. yyyy-mm-dd")]
        public DateTime EndDate { get; set; }

        [Usage(ApplicationAlias = "myExe")]
        public static IEnumerable<Example> Examples
        {
            get
            {
                var dateA = DateTime.Now.AddDays(-7);
                var dateB = DateTime.Now;
                yield return new Example("Normal scenario", new Options { StartDate = dateA, EndDate = dateB });
            }
        }
     }
```
command line:
```
Normal scenario:
myExe --endDate 3/20/2018 11:46:56 AM --startDate 3/13/2018 11:46:56 AM
```

desired:
myExe --endDate 2018-03-20  --startDate 2018-03-18

**Comment by tmtiwari** [comment]
Appears to be Duplicate of #502

**Comment by moh-hassan** [comment]
[online demo](https://dotnetfiddle.net/8gPgBK) to show the fix in v2.7+
